NDTensors/ext/NDTensorsMetalExt/adapt.jl

```diff
@@ -1,3 +1,4 @@
+using NDTensors.TypeParameterAccessors: TypeParameterAccessors
 NDTensors.cpu(e::Exposed{<:MtlArray}) = adapt(Array, e)
```

Comment:
Probably we should move `cpu` to `GPUArraysCoreExtensions`.

Reply:
Also I think that maybe `cpu` is a bad name for this operation because it doesn't account for the fact that there are many types of CPU Arrays like `Array`, `BlockSparseArray`, `DistributedArray` ... Potentially we want to make a `cpu(device_array, host_type = Array)` function in `GPUArraysCoreExtensions` to be more generic? 
This brings me to the thought that someone might try to do this 
```
cpu(Array, DistributedArray)
```
Which is still just a wrapper for `Adapt` but then the function wouldn't specifically match in `GPUArraysCoreExtensions` so we could do something like this instead `cpu(device_array::AbstractGPUArray, host_type = Array)`   Do you have any thoughts on this?

Reply:
I think `cpu` serves a clear purpose for single-processor calculations, to generically move data from the device to the host (the inverse of the functions `cu`, `mtl`, etc.). The same argument you make about it not being general enough for multi-processor environments can also be applied to the functions `cu`, `mtl`, etc. In terms of multi-processor environments, we will have to think about how all of those functions will need to be extended, but that is beyond the scope of this PR.